app/controllers/NirmsNumberController.scala

```diff
@@ -76,84 +73,37 @@ class NirmsNumberController @Inject() (
         )
   }
 
-  def onPageLoadUpdate: Action[AnyContent] =
+  def onPageLoadUpdate(mode: Mode): Action[AnyContent] =
     (identify andThen getData andThen requireData).async { implicit request =>
       traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
-        request.userAnswers.get(HasNirmsUpdatePage) match {
```

Comment:
This logic seemed needlessly complex so, I simplified to a minimal point where the AC's were met.

Reply:
Wow way nicer!